src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -402,281 +399,190 @@ bool OutputStateMachineEngine::ActionVt52EscDispatch(const VTID id, const gsl::s
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::CNL);
+        break;
     case CsiActionCodes::CPL_CursorPrevLine:
+        success = _dispatch->CursorPrevLine(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::CPL);
+        break;
     case CsiActionCodes::CHA_CursorHorizontalAbsolute:
     case CsiActionCodes::HPA_HorizontalPositionAbsolute:
+        success = _dispatch->CursorHorizontalPositionAbsolute(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::CHA);
+        break;
     case CsiActionCodes::VPA_VerticalLinePositionAbsolute:
+        success = _dispatch->VerticalLinePositionAbsolute(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::VPA);
+        break;
     case CsiActionCodes::HPR_HorizontalPositionRelative:
+        success = _dispatch->HorizontalPositionRelative(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::HPR);
+        break;
     case CsiActionCodes::VPR_VerticalPositionRelative:
-    case CsiActionCodes::ICH_InsertCharacter:
-    case CsiActionCodes::DCH_DeleteCharacter:
-    case CsiActionCodes::ECH_EraseCharacters:
-        success = _GetCursorDistance(parameters, distance);
+        success = _dispatch->VerticalPositionRelative(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::VPR);
         break;
-    case CsiActionCodes::HVP_HorizontalVerticalPosition:
     case CsiActionCodes::CUP_CursorPosition:
-        success = _GetXYPosition(parameters, line, column);
+    case CsiActionCodes::HVP_HorizontalVerticalPosition:
+        success = _dispatch->CursorPosition(parameters.at(0), parameters.at(1));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::CUP);
         break;
     case CsiActionCodes::DECSTBM_SetScrollingRegion:
-        success = _GetTopBottomMargins(parameters, topMargin, bottomMargin);
+        success = _dispatch->SetTopBottomScrollingMargins(parameters.at(0).value_or(0), parameters.at(1).value_or(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::DECSTBM);
+        break;
+    case CsiActionCodes::ICH_InsertCharacter:
+        success = _dispatch->InsertCharacter(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::ICH);
+        break;
+    case CsiActionCodes::DCH_DeleteCharacter:
+        success = _dispatch->DeleteCharacter(parameters.at(0));
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::DCH);
         break;
     case CsiActionCodes::ED_EraseDisplay:
+        success = parameters.for_each([&](const auto eraseType) {
+            return _dispatch->EraseInDisplay(eraseType);
+        });
+        TermTelemetry::Instance().Log(TermTelemetry::Codes::ED);
+        break;
     case CsiActionCodes::EL_EraseLine:
-        success = _GetEraseOperation(parameters, eraseType);
+        success = parameters.for_each([&](const auto eraseType) {

```

Comment:
Ahh, `VTParameters::for_each` will automatically `&&` this lambda's return value with the previous one, very clever.